website/docs/r/vpc_ipam_preview_next_cidr.html.markdown

```diff
@@ -0,0 +1,59 @@
+---
+subcategory: "VPC"
+layout: "aws"
+page_title: "AWS: aws_vpc_ipam_preview_next_cidr"
+description: |-
+  Previews a CIDR from an IPAM address pool.
+---
+
+# Resource: aws_vpc_ipam_preview_next_cidr
+
+Previews a CIDR from an IPAM address pool. Only works for private IPv4.
+
+## Example Usage
+
+Basic usage:
+
+```terraform
+data "aws_region" "current" {}
+
+resource "aws_vpc_ipam_preview_next_cidr" "example" {
+  ipam_pool_id   = aws_vpc_ipam_pool.example.id
+  netmask_length = 28
+
+  depends_on = [
+    aws_vpc_ipam_pool_cidr.example
+  ]
+}
+
+resource "aws_vpc_ipam_pool_cidr" "example" {
+  ipam_pool_id = aws_vpc_ipam_pool.example.id
+  cidr         = "172.2.0.0/16"
+}
+
+resource "aws_vpc_ipam_pool" "example" {
+  address_family = "ipv4"
+  ipam_scope_id  = aws_vpc_ipam.example.private_default_scope_id
+  locale         = data.aws_region.current.name
+}
+
+resource "aws_vpc_ipam" "example" {
+  operating_regions {
+    region_name = data.aws_region.current.name
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `ipam_pool_id` - (Required) The ID of the pool to which you want to assign a CIDR.
+* `netmask_length` - (Required) The netmask length of the CIDR you would like to preview from the IPAM pool.
```

Comment:
Its possible to pass without `netmask_length` if the ipam pool youre querying has `allocation_default_netmask_length` set.

https://github.com/hashicorp/terraform-provider-aws/blob/f445f5e1980a8b4a6581b2964b8ba70b80330128/internal/service/ec2/vpc_ipam_pool.go#L47

Reply:
Hi @drewmullen for this resource, I assumed that it won't have visibility into whether the `vpc_ipam_pool` has the `allocation_default_netmask_length` because if it doesn't the documentation mentions the following: `If there is no DefaultNetmaskLength allocation rule set on the pool, you must specify either the NetmaskLength or the CIDR.`. Since this resource is outputting the CIDR, it could be safer to just require the `netmask_length` to be set?

Reply:
Terraform should allow users to interact with the API as closely as possible. In the scenario youre mentioning, where useres do not setup the default and do not specify a netmask, they will get an API error because they're misconfiguring. This is actually how `aws_vpc` works as well

Reply:
Resolved in 4fe3f1703577ba7cf71cd727a4aedeb29b899e28